test/quickcheck/qc_catalog_merge_tool.cc

```diff
@@ -0,0 +1,204 @@
+    }
+  }
+
+  return spec;
+}
+
+// TODO(radu): Randomly modify the input spec
+DirSpec ModifySpec(const DirSpec& in) {
+  DirSpec out(in);
+
+  // Choose number of changes to perform
+  const auto num_changes = *rc::gen::arbitrary<size_t>();
+
+  for (auto i = 0u; i < num_changes; ++i) {
+    const auto change_type = *rc::gen::arbitrary<size_t>() % 4; // 0 == AddFile , 1 == AddDir etc.
+    switch (static_cast<ChangeType>(change_type)) {
+      case ChangeType::AddFile:
+      {
+        const auto dirs = out.GetDirs();
+        const auto parent_index = *rc::gen::arbitrary<size_t>() % dirs.size();
+        const auto parent = dirs[parent_index];
+        const auto file_name = "new_file" + std::to_string(i);
+        const auto file_size = *rc::gen::arbitrary<size_t>();
+        RC_ASSERT(out.AddFile(file_name, parent, hashes[0], file_size));
+      }
+      break;
+      case ChangeType::AddDir:
+      {
+        const auto dirs = out.GetDirs();
+        const auto parent_index = *rc::gen::arbitrary<size_t>() % dirs.size();
+        const auto parent = dirs[parent_index];
+        const auto dir_name = "new_dir" + std::to_string(i);
+        RC_ASSERT(out.AddDirectory(dir_name, parent, 1));
+      }
+      break;
+      case ChangeType::RemoveItem:
+      {
+        if (out.NumItems() > 0) {
+          const auto item_index = *rc::gen::arbitrary<size_t>() % out.NumItems();
+          size_t idx = 0;
+          auto it = out.items().begin();
+          while (idx < item_index) {
+            ++it;
+            ++idx;
+          }
+          out.RemoveItemRec(it->first);
+        }
+      }
+      break;
+      case ChangeType::ModifyFile:
+        //LogCvmfs(kLogCvmfs, kLogStdout, "ModifyFile");
+        break;
+      default:
+        LogCvmfs(kLogCvmfs, kLogStderr, "Unknown change type. Aborting.");
+        abort();
+        break;
+    }
+  }
+
+  return out;
+}
+
+receiver::Params MakeMergeToolParams(const std::string& name) {
+  receiver::Params params;
+
+  const std::string sandbox_root = GetCurrentWorkingDirectory();
+  const std::string stratum0 = sandbox_root + "/" + name + "_stratum0";
+  const std::string temp_dir = stratum0 + "/data/txn";
+
+  params.stratum0 = "file://" + stratum0;
+  params.spooler_configuration = "local," + temp_dir + "," + stratum0;
+  params.hash_alg = shash::kSha1;
+  params.compression_alg = zlib::kZlibDefault;
+  params.generate_legacy_bulk_chunks = false;
+  params.use_file_chunking = true;
+  params.min_chunk_size = 4194304;
+  params.avg_chunk_size = 8388608;
+  params.max_chunk_size = 16777216;
+  params.enforce_limits = false;
+  params.nested_kcatalog_limit = 0;
+  params.root_kcatalog_limit = 0;
+  params.file_mbyte_limit = 0;
+  params.use_autocatalogs = false;
+  params.max_weight = 0;
+  params.min_weight = 0;
+
+  return params;
+}
+
+}  // namespace
+
+class T_CatalogMergeTool : public ::testing::Test {};
+
+RC_GTEST_FIXTURE_PROP(T_CatalogMergeTool, InOut, ()) {
```

Comment:
A comment would we good with a brief high-level explanation about what is being tested.